src/renderers/native/ReactNativeStack.js

```diff
@@ -53,6 +44,25 @@ var ReactNative = {
   /* eslint-enable camelcase */
 
   unmountComponentAtNodeAndRemoveContainer: ReactNativeMount.unmountComponentAtNodeAndRemoveContainer,
+
+  __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED: {
+    // Used for Flow types
+    SyntheticEvent: require('SyntheticEvent'),
```

Comment:
I think this is wrong. You don't need to expose `SyntheticEvent` for Flow types. `SyntheticEvent` is a global in Flow (yes, Flow can be weird).